bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/ResourcesPlugin.java

```diff
@@ -32,12 +33,43 @@
 import org.osgi.framework.*;
 
 /**
- * The plug-in runtime class for the Resources plug-in.  This is
- * the starting point for all workspace and resource manipulation.
- * A typical sequence of events would be for a dependent plug-in
- * to call <code>ResourcesPlugin.getWorkspace()</code>.
- * Doing so would cause this plug-in to be activated and the workspace
- * (if any) to be loaded from disk and initialized.
+ * The plug-in runtime class for the Resources plug-in. This is the starting
+ * point for all workspace and resource manipulation. A typical sequence of
+ * events would be for a dependent plug-in to track the
+ * <code>org.eclipse.core.resources.IWorkspace</code> service. Doing so would
+ * cause this plug-in to be activated and the workspace (if any) to be loaded
+ * from disk and initialized.
+ *
+ * <p>
+ * Fetching the service could be performed in several ways:
+ * <ul>
+ * <li>For example in a Bundle/Plugin Activator:
+ *
+ * <pre>
+ * ServiceTracker<IWorkspace, IWorkspace> workspaceTracker = new ServiceTracker<>(bundleContext, IWorkspace.class,
+ * 		null);
+ * workspaceTracker.open();
+ * IWorkspace workspace = workspaceTracker.getService();
+ * </pre>
+ *
+ * </li>
+ * <li>Using declarative service with annotations:
+ *
+ * <pre>
+ * &#64;Reference
+ * private IWorkspace workspace;
```

Comment:
Why not on `getWorksapce` method documentation, since it's the method we aim at replacing? When this gets deprecated, it's the `getWorkpsace()` javadoc consumer will read, not `ResourcesPlugin`'s one.

Reply:
Because the ResourcesPlugin claims it is the way to acquire a workspace, actually I would aim to replace *all* calls to `ResourcesPlugin` and don't want to document each and every method with an example. For that reason as mentioned before I do not think adding "how to use osgi" examples at all as everyone will consider the documentation at different levels (the more interesting question would be why not place this at the `IWorkspace`....)

Reply:
We can slightly change the doc of `ResourcesPlugin`, but I believe doc on alternative to `getWorkspace` would still fit much better on `getWorkspace`

Reply:
I moved the example to `IWorkspace` and added a hint to `getWorkspace()`